pkg/instance/cache.go

```diff
@@ -0,0 +1,39 @@
+package instance
+
+import (
+	"time"
+
+	"github.com/cozy/cozy-stack/pkg/cache"
+)
+
+type instanceCache struct {
+	base cache.Cache
+}
+
+func (ic *instanceCache) Get(d string) *Instance {
+	var out Instance
+	ic.base.Get(d, &out)
+	if out.DocID == "" {
+		return nil
+	}
+	return &out
+}
+func (ic *instanceCache) Set(d string, i *Instance) {
+	ic.base.Set(d, i)
+}
+func (ic *instanceCache) Revoke(d string) {
+	ic.base.Del(d)
+}
+
+var globalCache *instanceCache
+
+func getCache() *instanceCache {
+
+	if globalCache == nil {
```

Comment:
You are setting a global variable from a function that can be called from several goroutines -> you need a mutex.